projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+// Comments ----------
+// I write a lot of comments, it has helped me the entire course to learn, understand and remember
+// They can all be removed at the end to clean it up
+
+func handleOKResponse(response *http.Response) {
+	// try to read the response body
+	responseBody, err := io.ReadAll(response.Body)
+
+	// use the defer keyword early to close the body regardless
+	defer response.Body.Close()
+
+	if err != nil {
+		// fmt.Fprintln(os.Stderr, "Parsing Response Body Failed. Error:", err)
+		// deliberately without explicit error message
+		fmt.Fprintln(os.Stderr, "Parsing Response Body Failed.")
+		return
+	}
+
+	// deliberately using os.Stdout (not fmt.Println) even though unnecessary (for the sake of the exercise)
+	fmt.Fprintln(os.Stdout, string(responseBody))
+}
+
+// note: returns an integer: 0 for fail || > 0 for success
+func parseRetryAfterHeader(retryAfterHeaderString string) int {
```

Comment:
Can you think of a way to re-implement this function using the examples for [error propagation](https://github.com/bazmurphy/immersive-go-course/tree/main/projects/output-and-error-handling#1-propagating-the-error-to-the-calling-function)?